nion/swift/LineGraphCanvasItem.py

```diff
@@ -59,90 +59,228 @@ def nice_label(value: float, precision: int) -> str:
+
+        min_specified = explicit_min_cal is not None
+        max_specified = explicit_max_cal is not None
+
+        if min_specified:
+            cal_min = typing.cast(float, explicit_min_cal)
+        else:
+            cal_min_opt: typing.Optional[float] = None
+            for xdata in xdata_list:
+                if xdata and xdata.data_shape[-1] > 0:
+                    arr = xdata.data if numpy.issubdtype(xdata.data.dtype, numpy.floating) else xdata.data.astype(float)
+                    if arr is not None:
+                        v = xdata.intensity_calibration.convert_to_calibrated_value(numpy.amin(arr))
+                        cal_min_opt = v if cal_min_opt is None else min(cal_min_opt, v)
+            cal_min = cal_min_opt if (cal_min_opt is not None and numpy.isfinite(cal_min_opt)) else 0.0
+
+        if max_specified:
+            cal_max = typing.cast(float, explicit_max_cal)
+        else:
+            cal_max_opt: typing.Optional[float] = None
+            for xdata in xdata_list:
+                if xdata and xdata.data_shape[-1] > 0:
+                    arr = xdata.data if numpy.issubdtype(xdata.data.dtype, numpy.floating) else xdata.data.astype(float)
+                    if arr is not None:
+                        v = xdata.intensity_calibration.convert_to_calibrated_value(numpy.amax(arr))
+                        cal_max_opt = v if cal_max_opt is None else max(cal_max_opt, v)
+            cal_max = cal_max_opt if (cal_max_opt is not None and numpy.isfinite(cal_max_opt)) else 0.0
+
+        if not min_specified and cal_min > 0.0:
+            cal_min = 0.0
+        if not max_specified and cal_max < 0.0:
+            cal_max = 0.0
+
+        if cal_min == cal_max:
+            cal_min -= 1.0
+            cal_max += 1.0
+
+        ticker = Geometry.LinearTicker(cal_min, cal_max)
+        disp_min = ticker.minimum if not min_specified else cal_min
+        disp_max = ticker.maximum if not max_specified else cal_max
+
+        return disp_min, disp_max, ticker
+
+    def make_ticker(self, display_min: float, display_max: float) -> Geometry.Ticker:
+        return Geometry.LinearTicker(display_min, display_max)
+
+    def to_display(self, value_cal: float) -> float:
+        return value_cal
+
+    def from_display(self, value_disp: float) -> float:
+        return value_disp
+
+    def transform_intensity_for_plot(self,xdata: DataAndMetadata.DataAndMetadata) -> DataAndMetadata.DataAndMetadata|None:
+        ic = xdata.intensity_calibration
+        if ic:
+            data = xdata.data if not xdata.is_data_rgb_type else Image.convert_to_grayscale(xdata.data)
+            cal = ic.offset + ic.scale * data
+            return DataAndMetadata.new_data_and_metadata(
+                cal,
+                intensity_calibration=Calibration.Calibration(units=ic.units),
+                dimensional_calibrations=xdata.dimensional_calibrations
+            )
+        return xdata
```

Comment:
Again, this is the wrong point to make the abstraction. There is enough common code between the two `transform_intensity_for_plot` implementations, that it is apparent it is the wrong abstraction.

Also, `ic`, as with `un` is a bad variable name. Can you leave the original descriptive variable names the same unless they are incorrect?